### ISSUE TITLE
org.tinylog.slf4j.TinylogLogger#isDebugEnabled() is bottle neck for Jetty's HttpOutput

### ISSUE THREAD

**Issue by simon04** [tinylog-org/tinylog#247]
**Describe the bug**
I'm running Jetty server with tinylog and slf4j-tinylog.

I've configured a custom DEBUG level for `foobar`:

```
level=INFO
level@foobar=DEBUG
writer1=console
writer1.format={date} {level} [{thread}] {class}.{method}(): {message}
```

`org.eclipse.jetty.server.HttpOutput` is very often calling `org.tinylog.slf4j.TinylogLogger#isDebugEnabled()` which turns out to be a bottle neck when serving a 17MB text file.

Here's a CPU profiling flamegraph I've generated using async-profiler: 

![image](https://user-images.githubusercontent.com/782446/161845567-f71d9b4c-e0c5-478d-8e08-4d87d2a68e4d.png)

**To Reproduce**
1. 
2.
3.

**Environment**
tinylog version: 2.5.0-M1
Java version: Azul 17

**Comment by pmwmedia** [comment]
I would like to analyze this use case. Could you provide a GitHub repository with a minimal reproducible example project? Can you reproduce the performance issue with Oracle Java? Which Jetty version do you use?

**Comment by simon04** [comment]
Here you are: https://github.com/simon04/tinylog/tree/issue-247

[profile.zip](https://github.com/tinylog-org/tinylog/files/8519286/profile.zip) has been obtained using

```
> /usr/lib/jvm/java-11-openjdk/bin/java -ea -agentpath:async-profiler-2.7-linux-x64/build/libasyncProfiler.so=start,file=/tmp/profile.html
Profiling started
2022-04-20 09:21:55 [main] org.eclipse.jetty.server.Server.doStart()
INFO: jetty-11.0.9; built: 2022-03-30T17:44:47.085Z; git: 243a48a658a183130a8c8de353178d154ca04f04; jvm 11.0.14.1+1
2022-04-20 09:21:55 [main] org.eclipse.jetty.server.handler.ContextHandler.doStart()
INFO: Started o.e.j.s.ServletContextHandler@16f7c8c1{/,null,AVAILABLE}
2022-04-20 09:21:55 [main] org.eclipse.jetty.server.AbstractConnector.doStart()
INFO: Started ServerConnector@97e1986{HTTP/1.1, (http/1.1)}{0.0.0.0:8080}
2022-04-20 09:21:55 [main] org.eclipse.jetty.server.Server.doStart()
INFO: Started Server@5b87ed94{STARTING}[11.0.9,sto=0] @644ms


> /usr/bin/curl localhost:8080 -o /dev/null
100 18.3M    0 18.3M    0     0  4318k      0 --:--:--  0:00:04 --:--:-- 4319k
```

**Comment by pmwmedia** [comment]
Thank you! I will investigate it.

**Comment by pmwmedia** [comment]
My first tests on Windows 10 without profiler:

**tinylog 2.5.0-M1**
```properties
level        = INFO
level@foobar = DEBUG
writer       = console
```
`Measure-Command { Invoke-WebRequest -Uri http://localhost:8080 }` takes **6.0559928** seconds

**Logback 1.2.11**
```xml
<configuration>
    <appender name="STDOUT" class="ch.qos.logback.core.ConsoleAppender">
        <encoder>
            <pattern>%d{HH:mm:ss.SSS} [%thread] %-5level %logger{36} - %msg%n</pattern>
        </encoder>
    </appender>
    <logger name="foobar" level="DEBUG" />
    <root level="INFO">
        <appender-ref ref="STDOUT" />
    </root>
</configuration>
```
`Measure-Command { Invoke-WebRequest -Uri http://localhost:8080 }` takes **0.6344634** seconds

Not such big difference between tinylog and Logback.

**Edit:** Enabled `$ProgressPreference = 'SilentlyContinue'`, otherwise the loading byte counter is the bottleneck.

**Comment by pmwmedia** [comment]
**tinylog 3.0-SNAPSHOT + SLF4J 2.0.0-alpha7**
```properties
level        = INFO
level@foobar = DEBUG
writer.type  = console
```
~`Measure-Command { Invoke-WebRequest -Uri http://localhost:8080 }` takes **1.2723528** seconds~
`Measure-Command { Invoke-WebRequest -Uri http://localhost:8080 }` takes **0.4198631** seconds (after commit 883d4ecdd391ec92c6604a630cb7ff7da7963685)

**Comment by pmwmedia** [comment]
The extremely frequent checking whether a severity level is enabled (while the severity level is only enabled for some classes) won't be anymore a bottleneck in tinylog 3.

**Comment by pmwmedia** [comment]
Fixed in latest snapshot of tinylog 3.0
